daemon/src/lib.rs

```diff
@@ -447,32 +447,34 @@ where
     }
 }
 
-#[derive(Debug, Copy, Clone, Display, PartialEq, Eq)]
-pub enum Environment {
-    Umbrel,
-    RaspiBlitz,
-    Citadel,
-    Start9,
-    MyNode,
-    Docker,
-    Binary,
-    Electron,
-    Test,
-    Legacy,
-    Unknown,
-}
+/// A struct defining our environment
+///
+/// We can run on all kinds of environment, hence this is just a wrapper around string.
+/// However, for backwards compatibility with <=0.6.x we need to support certain formats:
+/// `Docker`, `Umbrel`, `RaspiBlitz`, `Unknown`. For all other we format the string to lowercase.
+#[derive(Debug, Clone, Display, PartialEq, Eq)]
+pub struct Environment(String);
 
 impl Environment {
-    pub fn from_str_or_unknown(envvar_val: &str) -> Environment {
-        match envvar_val {
-            "umbrel" => Environment::Umbrel,
-            "raspiblitz" => Environment::RaspiBlitz,
-            "citadel" => Environment::Citadel,
-            "start9" => Environment::Start9,
-            "mynode" => Environment::MyNode,
-            "docker" => Environment::Docker,
-            "electron" => Environment::Electron,
-            _ => Environment::Unknown,
+    pub fn new(val: &str) -> Environment {
+        Self(Environment::parse_known_variances(val))
+    }
+
+    pub fn unknown() -> Environment {
+        Self("Unknown".to_string())
+    }
+
+    pub fn as_string(&self) -> String {
+        self.0.clone()
+    }
+
+    fn parse_known_variances(string: &str) -> String {
+        match string.to_lowercase().as_str() {
+            "docker" => "Docker".to_string(),
+            "umbrel" => "Umbrel".to_string(),
+            "raspiblitz" => "RaspiBlitz".to_string(),
```

Comment:
❓ It's not obvious why we need this. What part of the system needs the environment variable to be not-lowercase for these cases?

Reply:
Okay, I think the point is that we need to format certain variants differently, because old versions expect something other than lowercase.

Is this even worth it though? Only the `maker` is consuming this information to gather metrics. I would prefer it if we just "broke" this compatibility.

Reply:
If we break it we would need to update the identify protocol  and I didn't want to bother with this :D

Reply:
But I don't think that's true. The taker would just get an `Unknown` variant. And the taker doesn't even care about this information!

Reply:
Unfortunately not, he will get an error saying: can't parse XYZ, expected variants are `RaspiBlitz`, `Umbrel,`, etc...

Reply:
But I just tested removing this code and I get this on the taker:

```
2022-09-29 01:15:44  INFO xtra_actor_request{actor_type=daemon::identify::dialer::Actor message_type=daemon::identify::dialer::IdentifyMsgReceived}:xtra_message_handler:Handle IdentifyMsgReceived: daemon::identify::dialer: New identify message received peer_id=12D3KooWDjzHna3pNi1Bt1DoRfrpsBREykJKXDRDxXvhJNAdDZEk daemon_version=0.6.1 environment=Unknown wire_version=0.3.0
```

and this on the maker:

```
xtra_actor_request{actor_type=daemon::identify::dialer::Actor message_type=daemon::identify::dialer::IdentifyMsgReceived}:xtra_message_handler:Handle IdentifyMsgReceived: daemon::identify::dialer: New identify message received peer_id=12D3KooWBswfzW4GGePjyRGeZ9NHGNTCV4hXofcznpYEUbTKbsTg daemon_version=0.6.2 environment=Binary wire_version=0.3.0
```

Reply:
Mhm, thanks for trying. I'll look into this again.

Reply:
Got it, the only one we need to support is `Unknown`.